notation.go

```diff
@@ -89,23 +89,10 @@ func Sign(ctx context.Context, signer Signer, repo registry.Repository, signOpts
 	}
 
 	logger := log.GetLogger(ctx)
-	ref, err := orasRegistry.ParseReference(signOpts.ArtifactReference)
-	if err != nil {
-		return ocispec.Descriptor{}, err
-	}
-	if ref.Reference == "" {
-		return ocispec.Descriptor{}, errors.New("reference is missing digest or tag")
-	}
-	targetDesc, err := repo.Resolve(ctx, ref.Reference)
+	targetDesc, err := repo.Resolve(ctx, signOpts.ArtifactReference)
 	if err != nil {
 		return ocispec.Descriptor{}, fmt.Errorf("failed to resolve reference: %w", err)
 	}
-	if ref.ValidateReferenceAsDigest() != nil {
-		// artifactRef is not a digest reference
-		logger.Warnf("Always sign the artifact using digest(`@sha256:...`) rather than a tag(`:%s`) because tags are mutable and a tag reference can point to a different artifact than the one signed", ref.Reference)
-		logger.Infof("Resolved artifact tag `%s` to digest `%s` before signing", ref.Reference, targetDesc.Digest.String())
-	}
```

Comment:
We still need this warning.

Reply:
Consider
```go
	if signOpts.ArtifactReference != targetDesc.Digest.String() {
		ref, err := orasRegistry.ParseReference(signOpts.ArtifactReference)
		if err != nil || ref.ValidateReferenceAsDigest() != nil {
			// artifactRef is not a digest reference
			logger.Warnf("Always sign the artifact using digest(`@sha256:...`) rather than a tag(`:%s`) because tags are mutable and a tag reference can point to a different artifact than the one signed", ref.Reference)
			logger.Infof("Resolved artifact tag `%s` to digest `%s` before signing", ref.Reference, targetDesc.Digest.String())
		}
	}
```